File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        boolean minus = false;
        if (x < 0) {
            x = x * (-1);
            minus = true;
        }
        String intAsText = Integer.toString(x);
        StringBuffer sb = new StringBuffer(intAsText);
        String reversedInt = sb.reverse().toString();
        Integer parsedInteger = null;
        try {
            parsedInteger = Integer.parseInt(reversedInt);
        } catch (NumberFormatException e) {
            return 0;
        } 
        if (minus) {
            return (-1) * parsedInteger;
        }
        return parsedInteger;

    }
}

File: 0007-reverse-integer/0007-reverse-integer.kt
class Solution {
    fun reverse(x: Int): Int {
        var minus = false
        val intAsText: String
        if (x < 0) {
            intAsText = Integer.toString(x* (-1))
            minus = true
        } else {
            intAsText = Integer.toString(x)
        }
        
        val sb: StringBuffer = StringBuffer(intAsText)
        val reversedInt: String = sb.reverse().toString()
        var parsedInteger: Int
        try {
            parsedInteger = reversedInt.toInt()
        } catch (e: NumberFormatException) {
            return 0
        } 
        if (minus) {
            return (-1) * parsedInteger
        }
        return parsedInteger
    }
}